<a href="https://colab.research.google.com/github/lunacplayer6-ux/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/lunacplayer6-ux/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [ ]:
pwd

'/content/flyrank-ml-internship'

In [ ]:
!git clone https://github.com/lunacplayer6-ux/flyrank-ml-internship.git

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 124, done.
remote: Counting objects: 100% (124/124), done.
remote: Compressing objects: 100% (79/79), done.
remote: Total 124 (delta 39), reused 101 (delta 29), pack-reused 0 (from 0)
Receiving objects: 100% (124/124), 1.84 MiB | 10.96 MiB/s, done.
Resolving deltas: 100% (39/39), done.


In [ ]:
%cd flyrank-ml-internship

/content/flyrank-ml-internship/flyrank-ml-internship


In [ ]:
import os
print(os.getcwd())

/content/flyrank-ml-internship/flyrank-ml-internship


## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## Lane: Refresh / Content Opportunity Scoring

I chose the Refresh / Content Opportunity Scoring lane because it focuses on identifying pages that should be reviewed first for content updates. This is a practical business problem since reviewing every page manually is not possible. By analyzing search and content signals, I aim to build a ranking that helps prioritize which pages deserve attention first.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Decision
Help the content team decide which pages should be reviewed and updated first.

## Action
The SEO or content team will review the highest-ranked pages fSirst and update outdated or underperforming content.

## Cost of a wrong recommendation
If a page that does not need updates is prioritized, time and effort are wasted. If an important page is missed, it may continue to lose traffic and potential users.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("Average impressions (90d):", round(df["impressions_90d"].mean(), 2))
print("Average CTR:", round(df["ctr"].mean(), 2))

Total pages: 30000
Average impressions (90d): 5200.37
Average CTR: 0.51


### Observation

From the starter dataset:

- Total pages: **30,000**
- Average impressions (90 days): **5,200.37**
- Average CTR: **0.51**

The dataset is large enough to make manual page review difficult. Since pages receive different levels of traffic and engagement, prioritizing which pages should be refreshed first can help the content team focus on the most valuable opportunities.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## What I can claim

This project can identify pages that are good candidates for content refresh based on observed search and content signals. The results are intended to support decision-making, not replace human judgment.

## What I cannot claim

This project cannot guarantee that refreshing a page will improve rankings or traffic. It only identifies patterns in the available data and provides recommendations based on those observations.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.